src/lib/Validator/Constraints/UserPasswordValidator.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Validator\Constraints;
+
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use eZ\Publish\API\Repository\UserService;
+use eZ\Publish\API\Repository\Exceptions\InvalidArgumentException;
+use eZ\Publish\API\Repository\Values\User\User;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+
+/**
+ * Will check if logged user and password are match.
+ */
+class UserPasswordValidator extends ConstraintValidator
+{
+    /** @var UserService */
+    private $userService;
+
+    /** @var TokenStorageInterface */
+    private $tokenStorage;
+
+    /**
+     * @param UserService $userService
+     * @param TokenStorageInterface $tokenStorage
+     */
+    public function __construct(UserService $userService, TokenStorageInterface $tokenStorage)
+    {
+        $this->userService = $userService;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    /**
+     * Checks if the passed password exists for logged user.
+     *
+     * @param string $password The password that should be validated
+     * @param Constraint|UserPassword $constraint The constraint for the validation
+     *
+     * @throws \Symfony\Component\Validator\Exception\ConstraintDefinitionException
+     */
+    public function validate($password, Constraint $constraint)
+    {
+        if (null === $password || '' === $password) {
```

Comment:
I believe `empty()` will do just fine in this case.

Reply:
I disagree because `empty('0')` will also return true.

Reply:
I will give that one to you. ;)